Add support of compat.exp tests from GCC testsuite

kolerov opened the pull request:
Add fix_gcc_compat_multilib_flags function which fixes multilib flags for compat.exp tests from GCC testsuite. This function removes all multilib flags from board info and appends them to testglues's flags. It is necessary because compat.exp uses alternate compiler which may not support GCC's multilib flags (they must be passed manually to each compiler). However testglue must be compiled with those multilib flags so this function also appends them to testglue's flags using wrap_compile_flags variable. This function must be called last to let check_target_arc function detect ARC core by multilib flags correctly.

kolerov committed:
Add support of compat.exp tests from GCC testsuite

Add fix_gcc_compat_multilib_flags function which fixes multilib flags for
compat.exp tests from GCC testsuite. This function removes all multilib
flags from board info and appends them to testglues's flags. It is necessary
because compat.exp uses alternate compiler which may not support GCC's
multilib flags (they must be passed manually to each compiler). However
testglue must be compiled with those multilib flags so this function also
appends them to testglue's flags using wrap_compile_flags variable. This
function must be called last to let check_target_arc function detect ARC
core by multilib flags correctly.

Signed-off-by: Yuriy Kolerov <yuriy.kolerov@synopsys.com>

kolerov committed:
Use ARC_GCC_COMPAT_SUITE for turning on support of compat.exp tests

Turn on support of compat.exp tests from GCC testsuite if environment
variable ARC_GCC_COMPAT_SUITE == 1.

Signed-off-by: Yuriy Kolerov <yuriy.kolerov@synopsys.com>

anthony-kolesov commented:
This looks good to me, but there are a few more things that I need you to do:
- [x] Please update `dejagnu/baseboards/arc-nsim.exp` similarly to `arc-sim-nsimdrv.exp`
- [x] Update `dejagnu/example_run.sh` file, so that there would be a mention of "set ARC_GCC_COMPAT_SUITE to 1 if you are running compatibility tests".
- [x] Update [wiki page](https://github.com/foss-for-synopsys-dwc-arc-processors/toolchain/wiki/How-to-run-DejaGNU-tests) accordingly.

Then you can merge with arc-dev.

kolerov committed:
Add support of compat.exp tests from GCC testsuite to examples

These tests are intended to check compatibility between GNU and MetaWare
compilers. There is an example of configuration for ARC EM (little endian).

Signed-off-by: Yuriy Kolerov <yuriy.kolerov@synopsys.com>

kolerov commented:
I have updated examples, baseboards and wiki page. Is it fine now?

anthony-kolesov commented:
Excellent, please merge with arc-dev.
